aspnetcore/blazor/lifecycle.md

```diff
@@ -0,0 +1,173 @@
+---
+title: ASP.NET Core Blazor lifecycle
+author: guardrex
+description: Learn how to use Razor component lifecycle methods in ASP.NET Core Blazor apps.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 11/23/2019
+no-loc: [Blazor]
+uid: blazor/lifecycle
+---
+# ASP.NET Core Blazor lifecycle
+
+By [Luke Latham](https://github.com/guardrex) and [Daniel Roth](https://github.com/danroth27)
+
+The Blazor framework includes synchronous and asynchronous lifecycle methods. Override lifecycle methods to perform additional operations on components during component initialization and rendering.
```

Comment:
Instead of focusing purely on the lifecycle events I think we want this article to explain more broadly how components work, like how and when the render. We should walk the user through a day in the life a component in a Blazor app. The lifecycle events are then just specific points that you can plugin your specific behaviors. 

Things that I think should be covered:
- What happens during the initial render of the app
  - We probably need to address prerendering as part of this
- When components get instantiated
- When services get injected from DI
- What happens when a parent/root component gets rendered
- When do components render after the initial render
  - In response to UI events and explicit calls to StateHasChanged
- How long do components life for? When do they get disposed?

Reply:
... and cross-ref to Steve's original concepts, which include additional ideas for this topic ...

https://github.com/aspnet/AspNetCore.Docs/issues/14353

... and note that the idea here (this PR) was just to set up the new topic and then Steve would come in (or one of the other engineers) and modify the content from there.

Reply:
Ah ok, that's cool then.